vignettes/wrangling-linelist.Rmd

```diff
@@ -0,0 +1,128 @@
+:::
+
+## Simulate an outbreak
+
+To simulate an outbreak we will use the `sim_outbreak()` function from the {simulist} R package.
+
+::: {.alert .alert-info}
+If you are unfamiliar with the {simulist} package or the `sim_outbreak()` function [Get Started vignette](simulist.html) is a great place to start.
+:::
+
+First we load in some data that is required for the outbreak simulation. Data on epidemiological parameters and distributions are read from the {epiparameter} R package.
+
+```{r read-epidist}
+# create contact distribution (not available from {epiparameter} database)
+contact_distribution <- epidist(
+  disease = "COVID-19",
+  epi_dist = "contact distribution",
+  prob_distribution = "pois",
+  prob_distribution_params = c(mean = 2)
+)
+
+# create infectious period (not available from {epiparameter} database)
+infect_period <- epidist(
+  disease = "COVID-19",
+  epi_dist = "infectious period",
+  prob_distribution = "gamma",
+  prob_distribution_params = c(shape = 1, scale = 1)
+)
+
+# get onset to hospital admission from {epiparameter} database
+onset_to_hosp <- epidist_db(
+  disease = "COVID-19",
+  epi_dist = "onset to hospitalisation",
+  single_epidist = TRUE
+)
+
+# get onset to death from {epiparameter} database
+onset_to_death <- epidist_db(
+  disease = "COVID-19",
+  epi_dist = "onset to death",
+  single_epidist = TRUE
+)
+```
+
+The seed is set to ensure the output of the vignette is consistent. When using {simulist}, setting the seed is not required unless you need to simulate the same line list multiple times.
+
+```{r, set-seed}
+set.seed(123)
+```
+
+```{r, sim-outbreak}
+outbreak <- sim_outbreak(
+  contact_distribution = contact_distribution,
+  infect_period = infect_period,
+  prob_infect = 0.5,
+  onset_to_hosp = onset_to_hosp,
+  onset_to_death = onset_to_death
+)
+linelist <- outbreak$linelist
+contacts <- outbreak$contacts
+```
+
+## Removing a line list column {.tabset}
+
+Not every column in the simulated line list may be required for the use case at hand. In this example we will remove the `$ct_value` column. For instance, if we wanted to simulate an outbreak for which no laboratory testing (e.g Polymerase chain reaction, PCR, testing) was available and thus a Cycle threshold (Ct) value would not be known for confirmed cases.
+
+### Tidyverse
+
+```{r, rm-ct-col-tidyverse}
+# remove column by name
+linelist <- linelist %>%
+  select(!ct_value)
+linelist
+```
+
+### Base R
+
+```{r, rm-ct-col-base, eval=FALSE}
+# remove column by assigning it to NULL
+linelist$ct_value <- NULL
+
+# remove column by numeric column indexing
+# ct_value is column 12 (the last column)
+linelist <- linelist[, 1:11]
+
+# remove column by column name
+linelist <- linelist[, !colnames(linelist) == "ct_value"]
+linelist
+```
+
+```{r, rm-ct-col-base2, eval=TRUE, echo=FALSE}
+linelist
+```
```

Comment:
This is so that the `linelist` is printed, because the subsetting is already done in the tidyverse tab, I set the base R subsetting to `eval=FALSE`, so this is a separate chunk with `eval=TRUE` and `echo=FALSE` to print the `linelist` when on the Base R tab.

Reply:
In my experience, having chunks with `eval = FALSE` is a source of danger because they can easily go out of date without triggering any CI failures.

I've made an alternative suggestion but that's not a huge deal if you prefer to keep this for now.

Reply:
Thanks, I've made some modifications with your suggestions and using the previous commit suggestions in d7030763de7add28693ccee7aaaebd54fca767fb.

I don't think that having the `linelist` print three times in the Base R tab is needed, but as there is no text below this example for the moment I will merge it and tackle that once we extend the vignette.